ipopt-sys/build.rs

```diff
@@ -489,24 +489,24 @@ fn link(cnlp_install_path: PathBuf, link_info: LinkInfo, dynamic: bool) -> Resul
     if dynamic {
         println!("cargo:rustc-link-lib=dylib=ipopt");
     } else {
-        // Order is important here. The most core libs should appear last.
         println!("cargo:rustc-link-lib=static=ipopt");
-        for path in link_info.search_paths {
-            println!("cargo:rustc-link-search=native={}", path.display());
-        }
-        for (dep_type, lib) in link_info.libs {
-            let lib_type_str = match dep_type {
-                LibType::Dynamic => "dylib",
-                LibType::Static => "static",
-            };
-            println!("cargo:rustc-link-lib={}={}", lib_type_str, lib);
-        }
-        if cfg!(target_os = "macos") {
-            println!("cargo:rustc-link-lib=dylib=c++");
-        } else {
-            println!("cargo:rustc-link-lib=dylib=gfortran");
-            println!("cargo:rustc-link-lib=dylib=stdc++");
-        }
+    }
+        // Order is important here. The most core libs should appear last.
+    for path in link_info.search_paths {
+        println!("cargo:rustc-link-search=native={}", path.display());
+    }
+    for (dep_type, lib) in link_info.libs {
+        let lib_type_str = match dep_type {
+            LibType::Dynamic => "dylib",
+            LibType::Static => "static",
+        };
+        println!("cargo:rustc-link-lib={}={}", lib_type_str, lib);
+    }
+    if cfg!(target_os = "macos") {
+        println!("cargo:rustc-link-lib=dylib=c++");
+    } else {
+        println!("cargo:rustc-link-lib=dylib=gfortran");
+        println!("cargo:rustc-link-lib=dylib=stdc++");
```

Comment:
Ah interesting! In theory, if a lib (Ipopt here) is linked dynamically, there is no need to link ipopt dependencies (c++, stdc++, gfortran, etc.) since this linking is done when the dynamic ipopt lib is linked. It should only be needed when Ipopt is statically linked where the actual linking against Ipopt's dependencies is postponed to when an executable or a dynamic lib is being linked to Ipopt. I hope this makes some sense, unfortunately I don't have a good reference for this handy.

I suspect this change was motivated by the linking issues you had when linking with your system Ipopt lib. Is your system Ipopt lib a static lib or dynamic? If it's dynamic, are these extra linking steps (lines 493 to 509) still necessary or does it work after finding the `/usr/local` path?
If not, can you post the error output here?